### ISSUE TITLE
Add sample run information.

### ISSUE THREAD

**Issue by sholsapp** [yhenon/keras-frcnn#37]
To help people new to this, it would be great to add some sample run information, such as how long it takes to run on a particular system with particular hardware. Right now, I have no way to get a sense of how long training should take, and given infinite loops are/were a possibility, it would be nice to have a way to gut-check the running process. :)

**Comment by yhenon** [comment]
Good idea. I might just add a progress bar so people know where in the epoch things are? Would that solve the issue?

To give you an idea:
On my Titan X, it should be about 0.8s-0.9s per image at 600 pixel resolution for training RPN+classifier.
On CPU, it's very slow, about 8-9s/img

Regarding the possibility of infinite loops, I believe I've solved all those issues. I've yet to receive a report which showed an infinite loop on the latest codebase that is reproducible.

**Comment by sholsapp** [comment]
Just documenting the basic runtime information would be sufficient... e.g., should I wait 1 second, 1 minute, 1 hour, 1 day, 1 week, to train on something like the pascal VOC data?

> Regarding the possibility of infinite loops, I believe I've solved all those issues. I've yet to receive a report which showed an infinite loop on the latest codebase that is reproducible.

I understand this is a hard problem, but any programatic way to protect against this would be nice. Then again, if the user has a sense of how long some operation should take, then they can use common sense to learn something is wrong, too. :)

**Comment by yhenon** [comment]
Ok, definitely a good idea. It took me about 120k iterations for pascal voc (weights here: #23 ), so about 1.5 days.

**Comment by sholsapp** [comment]
Haha... yeah... definitely good to know... I didn't wait that long. :) Adding something to the README.md was be fantastic.

As a different request, can you update the README.md to have a note about where to get and how to add in those weight files? I think that speeds things up.

**Comment by mattdingmeng** [comment]
@yhenon @sholsapp  For each epoch, it takes ~1200 s on my GPU Titan Black. I used Pascal VOC 2007 and 2012, resnet 50.

**Comment by mattdingmeng** [comment]
@yhenon I tested the test_frcnn.py on my own images, which include many relatively small objects (vehicles). The detector perform very poor. Most of the small objects were missed. My images are around 400 by 500, and the vehicles are around 40 by 40 pixels.
Do you have any ideas to use your framework to detect these relatively small objects? In the original paper, the authors reported the capability to detect objects in large scale variance.

**Comment by sholsapp** [comment]
>  For each epoch, it takes ~1200 s on my GPU Titan Black.

That number's very different from the one @yhenon mentioned. Anyone know why?

Thanks @mattdingmeng

**Comment by yhenon** [comment]
@mattdingmeng 40px is quite small - try setting the anchor boxes to 64 pixels which might help. In general the detection rate on small objects is very low though - in the original paper they report a mAP < 0.5 for bottles for example.

@sholsapp @mattdingmeng 1200seconds/epoch -> 1.2 seconds/img is definitely slower than what I get (0.85s/img). However I'd have to know which backend, CUDA version, cuDNN being used, etc

**Comment by mattdingmeng** [comment]
@yhenon I use CUDA 8.0, cuDNN 5.1, backend Tensorflow.

**Comment by yhenon** [comment]
Can you confirm if you are on the latest version of the codebase?

**Comment by mattdingmeng** [comment]
@yhenon Tensorflow is v1.1 I remember. Keras is v2, but may not be the latest one. I download your code Wednesday and run it Thursday. It should be the latest one.

**Comment by yhenon** [comment]
I just realised you are using an Nvidia Titan black. My timings were done with a Titan X. The Titan X is more performant.

I just ran timings again and can confirm 0.82secs/img at train time (tested over 2k images)

**Comment by mattdingmeng** [comment]
@yhenon As you mentioned to change the size of anchor, do I need to re-train the model using the smaller anchor, or I just need to change the size of anchor in the test code?

**Comment by sholsapp** [comment]
@yhenon 

> It took me about 120k iterations for pascal voc (weights here: #23 ), so about 1.5 days.

Do iterations == epoch? So you trained w/ `--num_epochs=120000`? Based on @mattdingmeng numbers, that would take a lot longer than 1.5 days.

**Comment by wkelongws** [comment]
@sholsapp I believe @yhenon used 120 epochs (1000 iterations in each epoch). I tried the code on my own machine (TITAN X) and I got ~800s/epoch (0.8s/image) which is the similar speed performance with @yhenon

**Comment by alexattia** [comment]
Thanks for the information.  
Do you have some information about the number of pictures needed per class to perform well?
